packages/open-next/src/core/requestHandler.ts

```diff
@@ -37,11 +41,23 @@ export async function openNextHandler(
       }
       debug("internalEvent", internalEvent);
 
+      // These 3 will get overwritten by the routing handler if not using an external middleware
+      const internalHeaders = {
+        initialPath:
+          internalEvent.headers[INTERNAL_HEADER_INITIAL_PATH] ??
```

Comment:
nit: `internalEvent.headers` is repeated a lot, maybe create a `const headers = internalEvent.headers;` ?